### DISCUSSION TITLE
Using a MAX7219 LED Dot Matrix with ESP32 AudioKit

### DISCUSSION THREAD

**Discussion by xabadabazinho** [pschatzmann/arduino-audiokit#14] [Q&A] [answered]
Hi, 

I am trying  to use a MAX7219 LED Dot Matrix with ESP32 AudioKit; it works until I create an AudioKit object. Comment/uncomment the line creating the AudioKit object and the Dot Matrix starts/stops working.
I tried different pins for the SPI interface without sucess. 
Suspect audiokit board share some pins with other resources and make a conflict about it.
There is no error messages.
Has anyone sucessufully worked with this things ( SPI devices)  and an Audiokit Objec/ESP32A1S ?
Thanks for any help.

The code:

#include "Arduino.h"
#include <MD_Parola.h>
#include <SPI.h>
#include "AudioTools.h"
#include "AudioLibs/AudioKit.h"

#define HARDWARE_TYPE MD_MAX72XX::  FC16_HW
#define MAX_DEVICES	4 // number of MAX7219 ICs
#define CLK_PIN   18  // or SCK
#define CS_PIN    21  // or SS
#define DATA_PIN  23  // or MOSI

// SPI hardware interface
MD_Parola md = MD_Parola(HARDWARE_TYPE, CS_PIN, MAX_DEVICES);

////////////////comment this and all works great//////////////////////
AudioKit audiokit;
////////////////////////////////////////////////////////////////////////////
void setup()
{
      md.begin(); 
      Serial.begin(115200); 
}

void loop()
{
      md.print("text");
      delay(1000);
}

**Comment by pschatzmann** [comment]
SPI is used to access the SD drive. You could try to deactivate it with config. sd_active = false;
The sd is set up with       
SPI.begin(PIN_AUDIO_KIT_SD_CARD_CLK, PIN_AUDIO_KIT_SD_CARD_MISO, PIN_AUDIO_KIT_SD_CARD_MOSI, PIN_AUDIO_KIT_SD_CARD_CS);

Alternatively you might want to use another SPI port

**Reply by xabadabazinho** [reply]
Thanks for the reply. Tried  the code:
void setup(){
...
   auto cfg = audiokit.defaultConfig(AudioInput); 
    cfg.sd_active = false;    
    audiokit.begin(cfg);
SPI.begin(18, PIN_AUDIO_KIT_SD_CARD_MISO, 23 , 5);
...

Unfortunately, it doesn't work.

Please excuse-me, but I does not understood how to try "another SPI port".

**Comment by pschatzmann** [comment, marked as answer]
My mistake, I just noticed that the setupSPI() is also called in the constructor of AudioKit. So the only way would be to change the AUDIOKIT_SETUP_SD to 0 in the AudioKitSettings.h.

For your second question see https://www.esp32.com/viewtopic.php?t=20755
